### PR TITLE
vdk-core: Add 'method' to pre_ingest_process API

### PR DESCRIPTION
**Pull request by doks5**
This change adds a `method` argument to the pre_ingest_process() and post_ingest_process()
APIs, in order to allow plugin developers to add logic specific only to certain ingestion
methods.

Testing Done: CI/CD

### CODE REVIEW

File: projects/vdk-core/src/vdk/api/plugin/plugin_input.py [hunk 1]
```diff
@@ -387,6 +387,7 @@ def pre_ingest_process(
         target: Optional[str] = None,
         collection_id: Optional[str] = None,
         metadata: Optional[IngestionMetadata] = None,
+        method: Optional[str] = None,
```
Comment (antoniivanov) [inline projects/vdk-core/src/vdk/api/plugin/plugin_input.py:390]: Please add this to the post_ingest_process. 

They should be consistent.

**Review by antoniivanov — APPROVED** [review]
Looks good. thanks.

File: projects/vdk-core/src/vdk/api/plugin/plugin_input.py (hunk 1 above)
Comment (doks5) [reply projects/vdk-core/src/vdk/api/plugin/plugin_input.py:390]: Done. Thanks

**Review by dakodakov — APPROVED** [review]
LGTM

**Pull request merged**